ArgumentCountError in  AbstractListTypeToCTypeUpdate.php

SvenJuergens opened the pull request:
hi @linawolf 

thanks for the wizard it saves me a lot of upgrade time :) 
I wanted to use it directly in TYPO3 11 (before the upcoming update to 12). But then I got the following error,

(I deleted all caches including composer dump, unfortunately only the code customisation helped)

```
(1/1) ArgumentCountError
Too few arguments to function Linawolf\ListTypeMigration\Upgrades\AbstractListTypeToCTypeUpdate::__construct(), 0 passed in /var/www/html/public/typo3/sysext/core/Classes/Utility/GeneralUtility.php on line 3215 and exactly 1 expected
```

SvenJuergens committed:
ArgumentCountError in  AbstractListTypeToCTypeUpdate.php

linawolf commented:
Did you maybe override the constuctor without calling the parent constructor? It would probably be a good idea to switch to method injection instead of constructor injection.

Tests are failing

SvenJuergens committed:
[TASK] Switch to Method injection

SvenJuergens commented:
hi @linawolf, I more or less used this [example](https://docs.typo3.org/permalink/t3coreapi:plugins-list-type-migration-core-plugin-migration), for TYPO3 11 I added getIdentifier() and registered it in ext_localconf.php
I have now switched to method injection and used this [example](https://docs.typo3.org/permalink/t3coreapi:method-injection)

SvenJuergens committed:
Revert "[TASK] Switch to Method injection"

This reverts commit d071c05f430594db3f474d171352e06767517879.

SvenJuergens committed:
[TASK] PHP 8.4 fix

SvenJuergens commented:
@linawolf sorry, after switch to method injection got a Problem with $this->connectionPool is null.
So again back to constructor injection, with PHP 8.4 fix, seems php 7.4 fails with rector.php line 33 (named arguments are only allowed since PHP 8.0)

linawolf commented:
So is the extension using this missing its Configuration/Services.yaml maybe? I don't think the issue lies within this extension. DI works just well in TYPO3 11. You must however have a valid Services.yaml and delete all caches from install tool or console

linawolf commented:
The `??` operator did not yet exist in PHP 7.4 therefore your tests keep failing

SvenJuergens commented:
@linawolf Services.yaml is available :) 
all caches deleted, same problem.

But maybe you're right and it's just my installation, so I'm closing the pull request here :) 
sorry for the disturbance

because of the The null coalescing operator, he's been there since php 7 and used in TYPO3 Core itself: 
for example, so I thought it would be safe to use it.
https://github.com/TYPO3/typo3/blob/11.5/typo3/sysext/core/Classes/Page/AssetRenderer.php#L43

and the failed checks seems not related to the pull request:
PHP 7.4:
PHP Parse error:  syntax error, unexpected ':', expecting ')' in ./Build/rector/rector.php on line 33

PHP8.4:
PHP needs to be a minimum version of PHP 7.4.0 and maximum version of PHP 8.3.*.
At least that's what it says when you click on the links to the fail checks
